posts/inside-rust/2020-02-07-intro-rustc-self-profile.md

```diff
@@ -0,0 +1,269 @@
+---
+layout: post
+title: "Introduction to profiling rustc using the self profiler"
+author: Wesley Wiser
+description: "Learn how to use the -Zself-profile rustc flag"
+team: the self-profile working group <https://rust-lang.github.io/compiler-team/working-groups/self-profile/>
+---
+
+Over the last year, the [Self-Profile Working Group] has been building tools to profile `rustc`.
+This is part of the Compiler Team's ongoing efforts to improve `rustc`'s performance.
+In this post, we'll look at the tooling currently available and use them to profile an example crate's compile time.
+
+First, we'll download and build the `measureme` repository which provides tools to analyze self-profile trace data.
+
+```sh
+$ git clone https://github.com/rust-lang/measureme.git
+$ cd measureme
+$ cargo build --release --all
+```
+
+Now that we have our tools, let's download an example crate to profile its build.
+
+```sh
+$ cd ..
+$ git clone https://github.com/rust-lang/regex.git
+$ cd regex
+```
+
+We'll need to use a recent nightly compiler to get access to unsable `-Z` flags.
+
+```sh
+$ rustup override set nightly
+```
+
+If you haven't installed a nightly compiler before, this will download the nightly compiler for you.
+If you have, then update it to make sure you're on a recent version.
+
+```sh
+$ rustup update nightly
+```
+
+Now we can build it and tell `rustc` to profile the build of the `regex` crate.
+This will cause three new files to be created in the working directory which contain the profling data.
```

Comment:
I wonder if it would be better to encourage people to give a directory for emitting the data, e.g. `-Zself-profile=./profdata`